pkg/services/provisioning/values/values_test.go

```diff
@@ -241,6 +246,30 @@ func TestValues(t *testing.T) {
 	})
 }
 
+func TestValues_readFile(t *testing.T) {
+	type Data struct {
+		Val StringValue `yaml:"val"`
+	}
+
+	f, err := ioutil.TempFile(os.TempDir(), "file expansion *")
+	require.NoError(t, err)
+	file := f.Name()
+
+	defer func() {
+		require.NoError(t, os.Remove(file))
+	}()
+
+	const expected = "hello, world"
+	_, err = f.WriteString(expected)
+	require.NoError(t, err)
+	require.NoError(t, f.Close())
+
+	data := &Data{}
+	err = yaml.Unmarshal([]byte(fmt.Sprintf("val: $__file{%s}", file)), data)
+	assert.NoError(t, err)
```

Comment:
I think the test should halt immediately and not continue with assertions after an error.
```suggestion
	require.NoError(t, err)
```